common/inventory_profile.cpp

```diff
@@ -1807,4 +1822,157 @@ int16 EQ::InventoryProfile::FindFirstFreeSlotThatFitsItem(const EQ::ItemData *it
 		}
 	}
 	return 0;
-}
\ No newline at end of file
+}
+
+//This function has the same flaw as noted above
+//Helper functions for evolving items
+int16 EQ::InventoryProfile::HasEvolvingItem(uint64 evolve_unique_id, uint8 quantity, uint8 where)
+{
+	int16 slot_id = INVALID_INDEX;
+
+	//Altered by Father Nitwit to support a specification of
+	//where to search, with a default value to maintain compatibility
+
+	// Check each inventory bucket
+	if (where & invWhereWorn) {
+		slot_id = _HasEvolvingItem(m_worn, evolve_unique_id, quantity);
+		if (slot_id != INVALID_INDEX)
+			return slot_id;
+	}
+
+	if (where & invWherePersonal) {
+		slot_id = _HasEvolvingItem(m_inv, evolve_unique_id, quantity);
+		if (slot_id != INVALID_INDEX)
+			return slot_id;
+	}
+
+	if (where & invWhereBank) {
+		slot_id = _HasEvolvingItem(m_bank, evolve_unique_id, quantity);
+		if (slot_id != INVALID_INDEX)
+			return slot_id;
+	}
+
+	if (where & invWhereSharedBank) {
+		slot_id = _HasEvolvingItem(m_shbank, evolve_unique_id, quantity);
+		if (slot_id != INVALID_INDEX)
+			return slot_id;
+	}
+
+	if (where & invWhereTrading) {
+		slot_id = _HasEvolvingItem(m_trade, evolve_unique_id, quantity);
+		if (slot_id != INVALID_INDEX)
+			return slot_id;
+	}
+
+	// Behavioral change - Limbo is no longer checked due to improper handling of return value
+	if (where & invWhereCursor) {
+		// Check cursor queue
+		slot_id = _HasEvolvingItem(m_cursor, evolve_unique_id, quantity);
+		if (slot_id != INVALID_INDEX)
+			return slot_id;
+	}
```

Comment:
Let's use curlies here so scope is explicit.

Reply:
Reformatted as well.